server/database/dataGeneratorC.js

```diff
@@ -0,0 +1,97 @@
+// This is customized for Cassandra, whose data must contain generated productIds.
+
+const fs = require('fs');
+const faker = require('faker');
+const csv = require('csv-write-stream');
+
+const writer = csv();
+
+const numbers = [];
+for (let i = 1; i < 994; i += 1) {
+  numbers.push(i.toString().padStart(4, '0').concat('.jpg'));
+}
+
+const exceptions = ['0013.jpg', '0376.jpg', '0485.jpg', '0566.jpg', '0686.jpg', '0718.jpg', '0775.jpg', '0815.jpg', '0852.jpg', '0879.jpg', '0937.jpg', '0960.jpg'];
+
+const images = numbers.filter((item) => exceptions.indexOf(item) === -1);
+const imglen = images.length;
+
+const generate = (count, callback) => {
+  console.time('finished in');
+  const stream = fs.createWriteStream('cassandra-rainforest-related-items.csv');
+
+  stream.on('error', (err) => {
+    console.log(err);
+  });
+  stream.on('finish', () => {
+    console.log('Data finished writing to file.');
+  });
+
+  writer.pipe(stream);
+  // for (let i = 0; i < count; i += 1) {
+  let i = count;
+  const batchSize = Math.floor(count / 1000);
```

Comment:
Interesting how you decided to batch the process of generating your data. Do you happen to know how big of an impact this had on the speed/efficiency of your data generation? Have you initially tried it without batching?

Reply:
This didn't really batch my input - it just ensured that each section equal to that size had a different category, so that my category ID would increment at a slower pace than the rest of the file.